Feature/float2words sbu

SButko opened the pull request.

SButko committed:
[ADD] amount2words: setup, method

Adding new module for amount conversion
to words. Optional parameters can be used
to format string.

SButko committed:
[REF] float2words: rounding, tests

Renaming module to float2words. Adding some
unittests. Solving problem with decimal precision
and rounding. Adding more arguments for
wider usability.

oerp-odoo commented on float2words/float2words.py:
```diff
@@ -0,0 +1,43 @@
+        number_in_words (str): float number in words.
+
+    """
+    whole_number = int(number // 1)
```
Part where you split number into two part could have been moved to its own method: `_split_float` or something. It could be kind of useful on it own, because sometimes we might only need those parts and not to convert it to number. You can make a TODO, to refactor like this.

oerp-odoo commented on float2words/float2words.py:
```diff
@@ -0,0 +1,43 @@
+        '%.{}f'.format(precision) % rounded_number
+    ).split('.')[-1])
+    # add whitespace before suffix for better readability.
+    sfx1 = ' ' + sfx1 if sfx1 else sfx1
```
So we add whitespace every time? What if we use `,` which is default?, So it would look like `One hundred , thirteen cents` or `One hundred, thirteen cents`?

SButko replied:
We add whitespace before suffix only when suffix is passed. Default suffixes are equal to '' (it means - nothing, no whitespace) and default result would look like this: 'One hundred, thirteen cents' (here default connector==', ').

oerp-odoo replied:
My bad, I was thinking about suffix, but looking into connector default arg:).
Though I think we can make this a bit prettier.
You can dynamically create list and then use those values to join into string, like:
```python
s1 = 'blabla'
sfx1 = ''
s2 = 'lalal'
sfx2 = 'ct.'
' '.join([x for x in [s1, sfx1, s2, sfx2] if x])
```
Then we don't even need to use this clunky `+` to connect with suffixes and then in final variable too!

oerp-odoo commented on float2words/tests/test_float2words.py:
```diff
@@ -0,0 +1,63 @@
+        self.assertEqual(
+            self.Float2Words(13.23, 'en_US', sfx1='kg', sfx2='g',
+                             precision=3, connector=' and '),
+            'thirteen kg and two hundred and thirty g',
```
Why only english text tests? Do some for Lithuanian texts too. It won't hurt:)

SButko committed:
[REF] float2words: helper methods

Adding separate methods for float splitting
and number conbining with suffix.
Adding test case with Lituanian language.

oerp-odoo commented on float2words/float2words.py:
```diff
@@ -0,0 +1,76 @@
+    return number_in_words
+
+
+def _split_float(number, precision):
```
Incorrect methods order. Methods must go in order they depend on each other. `float2words` depends on `_split_float`, so `_split_float` must be defined first.

oerp-odoo commented on float2words/float2words.py:
```diff
@@ -0,0 +1,76 @@
+    return whole_part, decimal_part
+
+
+def _convert_and_combine(number, lang, suffix):
```
Why did you split this into separate method? It is kind of OK, but now here we have only two items:). So it does not look that useful to have iteration, when we actually check only one thing anyway.

SButko committed:
[REF] float2words: joining with sfx

Removing unnecessary helper method.
Fixing joining numbers with suffixes.
Changing method sequence by dependency.

oerp-odoo commented on float2words/float2words.py:
```diff
@@ -55,22 +27,28 @@ def _split_float(number, precision):
-            suffix
-        ] if item
+    whole_part, decimal_part = _split_float(number, precision)
+    number_in_words = ''.join([
```
Looks, good, just missing comment explaining why we are doing this, the way we are doing:)

SButko committed:
[REF] float2words: comments

Adding some comments explaining why
and how we are joining number with suffix.
